include/boost/geometry/util/math.hpp

```diff
@@ -796,6 +820,17 @@ inline Result rounding_cast(T const& v)
     return detail::rounding_cast<Result, T>::apply(v);
 }
 
+/*
+\brief Short utility to divide. If the division is integer, it rounds the division
+       to the nearest value, without using floating point calculations
+\ingroup utility
+*/
+template <typename T>
+inline T divide(T const& n, T const& d)
+{
+    return detail::divide<T>::apply(n, d);
+}
```

Comment:
I'd like to propose to move away from this implementation technique (function calling struct member function) and use more modern one instead. This way there is one less function call per utility, which means smaller call trees and therefore simpler compilation error messages, maybe shorter compilation times? I'm proposing to do this in details, utilities such as this one, etc. The tags dispatching of algorithms would stay as it is. So this implementation could be replaced with:
```
template <typename T, std::enable_if_t<! std::is_integral<T>::value, int> = 0>
inline T divide(T const& n, T const& d)
{
    return ...
}

template <typename T, std::enable_if_t<std::is_integral<T>::value, int> = 0>
inline T divide(T const& n, T const& d)
{
    return ...
}
```

Reply:
> I'd like to propose to move away from this implementation technique (function calling struct member function) and use more modern one instead. This way there is one less function call per utility, which means smaller call trees and therefore simpler compilation error messages, maybe shorter compilation times? I'm proposing to do this in details, utilities such as this one, etc. The tags dispatching of algorithms would stay as it is. So this implementation could be replaced with:
> 
> ```
> template <typename T, std::enable_if_t<! std::is_integral<T>::value, int> = 0>
> inline T divide(T const& n, T const& d)
> {
>     return ...
> }
> 
> template <typename T, std::enable_if_t<std::is_integral<T>::value, int> = 0>
> inline T divide(T const& n, T const& d)
> {
>     return ...
> }
> ```

I don't agree with this. Our architecture is tag dispatching and specialization, we've that all over the place, and I don't feel to rebuild that all.
This is made consistent with the other code, in the math unit - but basically everywhere.

Maybe, if C++20 is there, we can make use of concepts, and we might make another choice, but that will first need some research (especially about error messages in case of errors).